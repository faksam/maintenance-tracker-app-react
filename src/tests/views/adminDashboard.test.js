import React from 'react';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import ConnectedDashboard, { Dashboard } from '../../app/views/AdminDashboard';
import mockData from '../__mocks__/mockData';

const props = {
  fullname: 'fakunlesamuel@gmail.com',
  email: 'fakunlesamuel@gmail.com',
  password: 'Lifeisarace',
  confirmPassword: 'Lifeisarace',
  handleSignIn: jest.fn(),
  handleSignUp: jest.fn(),
  handleChange: jest.fn(),
  handleSignOut: jest.fn(),
  hideModalRequest: jest.fn(),
  signoutUserAction: jest.fn(),
  loadRequestsAction: jest.fn(),
  requests: mockData.data,
  request: mockData.data[0],
  history: {},
  user: {
    username: 'joeeasy',
    email: 'joeeasy@gmail.com',
    role: 'Admin',
    id: '6bf54873-6033-4ab0-855d-336a34e1dfbc',
    iat: 1540985284931
  },
};
const initialState = {
  auth: {
    isAuthenticated: true,
    error: {},
    user: {
      username: 'joeeasy',
      email: 'joeeasy@gmail.com',
      role: 'Admin',
      id: '6bf54873-6033-4ab0-855d-336a34e1dfbc',
      iat: 1540985284931
    }
  },
  requestReducer: {
    requests: mockData.data,
    request: mockData.data[0],
  }

};
const mockStore = configureMockStore([thunk]);
const store = mockStore(initialState);

describe('Dashboard Component', () => {
  it('should render without throwing an error', () => {
    const wrapper = shallow(<Dashboard store={store} {...props} />);
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
  it('should set state to props', () => {
    const wrapper = shallow(<Dashboard store={store} {...props} />);
    expect(wrapper.state()).toEqual({
      request: {},
      showViewRequest: false
    });
  });
  it('should render without throwing an error when connected to store', () => {
    const wrapper = shallow(<ConnectedDashboard store={store} {...props} />);
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
  it('should hide view modal', () => {
    const wrapper = shallow(<Dashboard {...props} />);
    const instance = wrapper.instance();
    const e = {
      preventDefault: jest.fn()
    };
    instance.viewClickedRequest({}, e);
    instance.hideModalRequest();
  });
});
