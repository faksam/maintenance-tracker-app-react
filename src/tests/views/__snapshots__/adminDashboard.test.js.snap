// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component should render without throwing an error 1`] = `
<div>
  <div
    className="main-container"
  >
    <div
      className="row"
    >
      <section
        className="navigation"
      >
        <div
          className="nav-container"
        >
          <div
            className="brand"
          >
            <a
              className="dashboard-logo"
              href="./homepage.html"
            >
              <img
                alt="Logo"
                className="logo-font"
                src={Object {}}
              />
               
              Â  M-T-A
            </a>
          </div>
          <nav>
            <div
              className="nav-mobile"
            >
              <a
                href="#!"
                id="nav-toggle"
              >
                <span />
              </a>
            </div>
            <ul
              className="nav-list"
            >
              <li
                className="hide-div"
                id="adminhomepage"
              >
                <a
                  href="./adminhomepage.html"
                >
                  Home
                </a>
              </li>
              <li>
                <a
                  href="#!"
                >
                  Account
                </a>
                <ul
                  className="nav-dropdown"
                >
                  <li>
                    <a
                      href="./homepage.html"
                    >
                      My Requests
                    </a>
                  </li>
                  <li>
                    <a
                      href="./user_account.html"
                    >
                      My Account
                    </a>
                  </li>
                </ul>
              </li>
              <li>
                <a
                  id="signoutButton"
                  onClick={[Function]}
                >
                  Sign Out
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </section>
    </div>
    <div
      className="container"
    >
      <br />
      <div
        className="row"
      >
        <h2
          className="center"
        >
          All Requests
        </h2>
        <div
          className="row"
        >
          <div
            className="col-m col-1-6-m"
          >
            <a>
              <button
                id="createRequestButton"
                type="button"
              >
                <i
                  className="material-icons"
                >
                  add
                </i>
              </button>
            </a>
          </div>
          <div
            className="col-m col-1-2-m"
          >
            <input
              id="filterRequestInput"
              placeholder="Search"
            />
          </div>
          <div
            className="col-m col-1-3-m"
          >
            <select
              id="filterByStatus"
            >
              <option
                value="All"
              >
                All
              </option>
              <option
                value="New"
              >
                New
              </option>
              <option
                value="Pending"
              >
                Approved
              </option>
              <option
                value="Disapproved"
              >
                Disapproved
              </option>
              <option
                value="Resolved"
              >
                Resolved
              </option>
            </select>
          </div>
        </div>
        <table
          id="requestTable"
        >
          <thead>
            <tr
              className="header"
            >
              <th>
                Title
              </th>
              <th
                colSpan="2"
              >
                Description
              </th>
              <th>
                Status
              </th>
              <th
                colSpan="2"
              >
                Date
              </th>
              <th
                colSpan="2"
              >
                Edit/Delete
              </th>
            </tr>
          </thead>
          <tbody
            id="requestTableBody"
          >
            <RequestRow
              key="16"
              request={
                Object {
                  "comment": null,
                  "date": "2018-10-31T13:07:52.125Z",
                  "description": "Who am I Who am I Who am I Who am I Who am I",
                  "id": 16,
                  "status": "New",
                  "title": "Who am I",
                  "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
                }
              }
              viewClickedRequest={[Function]}
            />
            <RequestRow
              key="15"
              request={
                Object {
                  "comment": null,
                  "date": "2018-10-31T13:06:23.714Z",
                  "description": "Imela Frank E Imela Frank E Imela Frank E Imela Frank E Imela Frank E Imela Frank E Imela Frank E",
                  "id": 15,
                  "status": "New",
                  "title": "Imela Frank E",
                  "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
                }
              }
              viewClickedRequest={[Function]}
            />
            <RequestRow
              key="14"
              request={
                Object {
                  "comment": null,
                  "date": "2018-10-31T13:03:37.098Z",
                  "description": "New Request New New Request New New Request New New Request New New Request New New Request New New Request New",
                  "id": 14,
                  "status": "New",
                  "title": "New Request New",
                  "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
                }
              }
              viewClickedRequest={[Function]}
            />
            <RequestRow
              key="13"
              request={
                Object {
                  "comment": null,
                  "date": "2018-10-31T12:53:59.199Z",
                  "description": "New York Times New York Times New York Times New York Times",
                  "id": 13,
                  "status": "New",
                  "title": "New York Times pa",
                  "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
                }
              }
              viewClickedRequest={[Function]}
            />
            <RequestRow
              key="12"
              request={
                Object {
                  "comment": null,
                  "date": "2018-10-31T12:52:07.083Z",
                  "description": "New York Times New York Times New York Times New York Times New York Times New York Times",
                  "id": 12,
                  "status": "New",
                  "title": "New York Times friday",
                  "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
                }
              }
              viewClickedRequest={[Function]}
            />
            <RequestRow
              key="11"
              request={
                Object {
                  "comment": null,
                  "date": "2018-10-31T12:44:17.617Z",
                  "description": "New Request esm New Request esm New Request esm New Request esm New Request esm New Request esm",
                  "id": 11,
                  "status": "New",
                  "title": "New Request esm",
                  "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
                }
              }
              viewClickedRequest={[Function]}
            />
            <RequestRow
              key="10"
              request={
                Object {
                  "comment": null,
                  "date": "2018-10-31T12:41:25.920Z",
                  "description": "New York Times New York Times New York Times New York Times New York Times New York Times New York Times New York Times New York Times",
                  "id": 10,
                  "status": "New",
                  "title": "New York Times 7",
                  "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
                }
              }
              viewClickedRequest={[Function]}
            />
            <RequestRow
              key="9"
              request={
                Object {
                  "comment": null,
                  "date": "2018-10-31T12:40:28.382Z",
                  "description": "Oluwa miOluwa miOluwa miOluwa miOluwa miOluwa miOluwa miOluwa miOluwa mi",
                  "id": 9,
                  "status": "New",
                  "title": "Oluwa mi",
                  "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
                }
              }
              viewClickedRequest={[Function]}
            />
            <RequestRow
              key="8"
              request={
                Object {
                  "comment": null,
                  "date": "2018-10-31T12:27:36.724Z",
                  "description": "showRequestForm New York Times44 showRequestForm New York Times44 showRequestForm New York Times44 showRequestForm New York Times44",
                  "id": 8,
                  "status": "New",
                  "title": "New York Times44",
                  "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
                }
              }
              viewClickedRequest={[Function]}
            />
            <RequestRow
              key="7"
              request={
                Object {
                  "comment": null,
                  "date": "2018-10-31T12:25:34.133Z",
                  "description": "New York Times New York Times New York Times",
                  "id": 7,
                  "status": "New",
                  "title": "New York Times 2",
                  "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
                }
              }
              viewClickedRequest={[Function]}
            />
            <RequestRow
              key="6"
              request={
                Object {
                  "comment": null,
                  "date": "2018-10-31T12:24:26.675Z",
                  "description": "New York Times New York Times New York Times New York Times New York Times",
                  "id": 6,
                  "status": "New",
                  "title": "New York Times",
                  "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
                }
              }
              viewClickedRequest={[Function]}
            />
            <RequestRow
              key="5"
              request={
                Object {
                  "comment": null,
                  "date": "2018-05-24T23:00:00.000Z",
                  "description": "All the Office Chairs Are squeaky in my office and one is broken. When will it be fixed and repaired. It is very important.",
                  "id": 5,
                  "status": "New",
                  "title": "Office Chairs Are All squeaky",
                  "userid": "1e8a9783-40a7-453b-b3c3-9f5c8bab7988",
                }
              }
              viewClickedRequest={[Function]}
            />
            <RequestRow
              key="4"
              request={
                Object {
                  "comment": null,
                  "date": "2018-04-23T23:00:00.000Z",
                  "description": "One of the Office Printers in my office is broken. When will it be fixed and repaired. It is very important.",
                  "id": 4,
                  "status": "New",
                  "title": "Office Printer Not Working",
                  "userid": "1e8a9783-40a7-453b-b3c3-9f5c8bab7988",
                }
              }
              viewClickedRequest={[Function]}
            />
            <RequestRow
              key="3"
              request={
                Object {
                  "comment": null,
                  "date": "2018-04-22T23:00:00.000Z",
                  "description": "Please My laptop is emitting smoke and over heating. When will it be fixed and repaired. It is very important.",
                  "id": 3,
                  "status": "New",
                  "title": "Laptop Over-Heating",
                  "userid": "041b755d-3651-4571-92b2-f28bcbf8adcc",
                }
              }
              viewClickedRequest={[Function]}
            />
            <RequestRow
              key="2"
              request={
                Object {
                  "comment": null,
                  "date": "2018-04-20T23:00:00.000Z",
                  "description": "One of the Monitors in my office is broken. When will it be fixed and repaired. It is very important.",
                  "id": 2,
                  "status": "New",
                  "title": "Broken Monitor",
                  "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
                }
              }
              viewClickedRequest={[Function]}
            />
            <RequestRow
              key="1"
              request={
                Object {
                  "comment": null,
                  "date": "2018-04-19T23:00:00.000Z",
                  "description": "One of the desks in my office is broken. When will it be fixed and repaired. It is very important.",
                  "id": 1,
                  "status": "New",
                  "title": "Broken Desk",
                  "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
                }
              }
              viewClickedRequest={[Function]}
            />
          </tbody>
        </table>
      </div>
       
      <br />
      <br />
    </div>
    <br />
    <br />
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="pagination"
        >
          <section
            id="paginationSection"
          >
            <h2>
              Jump to page:
            </h2>
          </section>
        </div>
      </div>
    </div>
    <div
      className="footer"
    >
      <div
        className="row"
      >
        <p>
          Â© 2018 Copyright M-T-A by
          <a
            href="http://www.fakunlesamuel.com"
            target="_blank"
          >
            FakSam
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Dashboard Component should render without throwing an error when connected to store 1`] = `
<Dashboard
  auth={true}
  confirmPassword="Lifeisarace"
  email="fakunlesamuel@gmail.com"
  error={Object {}}
  fullname="fakunlesamuel@gmail.com"
  handleChange={[MockFunction]}
  handleSignIn={[MockFunction]}
  handleSignOut={[MockFunction]}
  handleSignUp={[MockFunction]}
  hideModalRequest={[MockFunction]}
  history={Object {}}
  loadRequestsAction={[Function]}
  password="Lifeisarace"
  request={
    Object {
      "comment": null,
      "date": "2018-10-31T13:07:52.125Z",
      "description": "Who am I Who am I Who am I Who am I Who am I",
      "id": 16,
      "status": "New",
      "title": "Who am I",
      "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
    }
  }
  requests={
    Array [
      Object {
        "comment": null,
        "date": "2018-10-31T13:07:52.125Z",
        "description": "Who am I Who am I Who am I Who am I Who am I",
        "id": 16,
        "status": "New",
        "title": "Who am I",
        "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
      },
      Object {
        "comment": null,
        "date": "2018-10-31T13:06:23.714Z",
        "description": "Imela Frank E Imela Frank E Imela Frank E Imela Frank E Imela Frank E Imela Frank E Imela Frank E",
        "id": 15,
        "status": "New",
        "title": "Imela Frank E",
        "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
      },
      Object {
        "comment": null,
        "date": "2018-10-31T13:03:37.098Z",
        "description": "New Request New New Request New New Request New New Request New New Request New New Request New New Request New",
        "id": 14,
        "status": "New",
        "title": "New Request New",
        "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
      },
      Object {
        "comment": null,
        "date": "2018-10-31T12:53:59.199Z",
        "description": "New York Times New York Times New York Times New York Times",
        "id": 13,
        "status": "New",
        "title": "New York Times pa",
        "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
      },
      Object {
        "comment": null,
        "date": "2018-10-31T12:52:07.083Z",
        "description": "New York Times New York Times New York Times New York Times New York Times New York Times",
        "id": 12,
        "status": "New",
        "title": "New York Times friday",
        "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
      },
      Object {
        "comment": null,
        "date": "2018-10-31T12:44:17.617Z",
        "description": "New Request esm New Request esm New Request esm New Request esm New Request esm New Request esm",
        "id": 11,
        "status": "New",
        "title": "New Request esm",
        "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
      },
      Object {
        "comment": null,
        "date": "2018-10-31T12:41:25.920Z",
        "description": "New York Times New York Times New York Times New York Times New York Times New York Times New York Times New York Times New York Times",
        "id": 10,
        "status": "New",
        "title": "New York Times 7",
        "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
      },
      Object {
        "comment": null,
        "date": "2018-10-31T12:40:28.382Z",
        "description": "Oluwa miOluwa miOluwa miOluwa miOluwa miOluwa miOluwa miOluwa miOluwa mi",
        "id": 9,
        "status": "New",
        "title": "Oluwa mi",
        "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
      },
      Object {
        "comment": null,
        "date": "2018-10-31T12:27:36.724Z",
        "description": "showRequestForm New York Times44 showRequestForm New York Times44 showRequestForm New York Times44 showRequestForm New York Times44",
        "id": 8,
        "status": "New",
        "title": "New York Times44",
        "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
      },
      Object {
        "comment": null,
        "date": "2018-10-31T12:25:34.133Z",
        "description": "New York Times New York Times New York Times",
        "id": 7,
        "status": "New",
        "title": "New York Times 2",
        "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
      },
      Object {
        "comment": null,
        "date": "2018-10-31T12:24:26.675Z",
        "description": "New York Times New York Times New York Times New York Times New York Times",
        "id": 6,
        "status": "New",
        "title": "New York Times",
        "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
      },
      Object {
        "comment": null,
        "date": "2018-05-24T23:00:00.000Z",
        "description": "All the Office Chairs Are squeaky in my office and one is broken. When will it be fixed and repaired. It is very important.",
        "id": 5,
        "status": "New",
        "title": "Office Chairs Are All squeaky",
        "userid": "1e8a9783-40a7-453b-b3c3-9f5c8bab7988",
      },
      Object {
        "comment": null,
        "date": "2018-04-23T23:00:00.000Z",
        "description": "One of the Office Printers in my office is broken. When will it be fixed and repaired. It is very important.",
        "id": 4,
        "status": "New",
        "title": "Office Printer Not Working",
        "userid": "1e8a9783-40a7-453b-b3c3-9f5c8bab7988",
      },
      Object {
        "comment": null,
        "date": "2018-04-22T23:00:00.000Z",
        "description": "Please My laptop is emitting smoke and over heating. When will it be fixed and repaired. It is very important.",
        "id": 3,
        "status": "New",
        "title": "Laptop Over-Heating",
        "userid": "041b755d-3651-4571-92b2-f28bcbf8adcc",
      },
      Object {
        "comment": null,
        "date": "2018-04-20T23:00:00.000Z",
        "description": "One of the Monitors in my office is broken. When will it be fixed and repaired. It is very important.",
        "id": 2,
        "status": "New",
        "title": "Broken Monitor",
        "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
      },
      Object {
        "comment": null,
        "date": "2018-04-19T23:00:00.000Z",
        "description": "One of the desks in my office is broken. When will it be fixed and repaired. It is very important.",
        "id": 1,
        "status": "New",
        "title": "Broken Desk",
        "userid": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
      },
    ]
  }
  signoutUserAction={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  user={
    Object {
      "email": "joeeasy@gmail.com",
      "iat": 1540985284931,
      "id": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
      "role": "Admin",
      "username": "joeeasy",
    }
  }
/>
`;
